apps/meteor/client/views/room/composer/messageBox/MessageBox.tsx

```diff
@@ -454,10 +695,10 @@ const MessageBox = ({
 								<MessageComposerAction
 									aria-label={t('Send')}
 									icon='send'
-									disabled={!canSend || (!typing && !isEditing)}
+									disabled={!canSend || (!typing && !isEditing && !(filesToUpload.length > 0))}
 									onClick={handleSendMessage}
-									secondary={typing || isEditing}
-									info={typing || isEditing}
+									secondary={typing || isEditing || filesToUpload.length > 0}
+									info={typing || isEditing || filesToUpload.length > 0}
```

Comment:
Would you mind to convert those `filesToUpload.length > 0` to a state `isUploading`? DRY and reduce calculations

Reply:
done sir added the state isUploading